landoapi/decorators.py

```diff
@@ -0,0 +1,90 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import functools
+import logging
+
+from connexion import problem
+from flask import request
+from netaddr import IPAddress, IPNetwork
+
+logger = logging.getLogger(__name__)
+
+
+def get_ips():
+    if len(request.access_route) > 0:
+        return list(request.access_route)
+    return [request.remote_addr]
+
+
+def limit_ips(
+    whitelist_ip,
+    inbound_proxies,
+    log_tag,
+    log_msg='Access denied',
+    err_title='Not authorized',
+    err_msg='Access denied'
+):
+    """Allow only whitelisted IPs.
+
+    Args:
+        whitelist_ip: list
+            allowed Networks ['127.0.0.1', '192.168.0.1/8']
+        inbound_proxies: list
+            allowed proxies IP addresses
+        log_tag: string
+            tag the warning in logs
+        log_msg: string
+            Message to log
+        err_title: string
+            Title of the message to respond to the client
+        err_message: string
+            Body of the message to respond to the client
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+            ips = get_ips()
+            ip = ips.pop(0)
+
+            valid_proxy = False
+            if (len(ips) > 0 and not set(inbound_proxies) & set(ips)):
+                logger.warning(
+                    {
+                        'remote_addr': ip,
+                        'msg': 'Wrong proxy. %s' % log_msg,
+                    }, log_tag
+                )
+                return problem(
+                    403,
+                    err_title,
+                    err_msg,
+                    type=
+                    'https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403'
+                )
```

Comment:
this doesn't make much sense.

the whole deal with configuring for proxies is they are used to indicate which ip addresses can be trusted in `x-forwarded-for`. note the bugzilla code i provided as a reference on irc is part of it's util function that returns the client's ip address.

remove this as well as the logic for iterating over each ip in the `x-forwarded-for` header.  instead write a custom `get_remote_address` function that uses the configured list of trusted proxies to determine the remote endpoint's address.  use that function any place you need to get the client's ip address.

    trusted_proxies = {'127.0.0.1'}
    route = request.access_route + [request.remote_addr]
    remote_addr = next((addr for addr in reversed(route) 
                        if addr not in trusted_proxies), request.remote_addr)